pkg/controllers/mcs/endpointslice_controller.go

```diff
@@ -54,21 +55,20 @@ func (c *EndpointSliceController) Reconcile(ctx context.Context, req controllerr
 	work := &workv1alpha1.Work{}
 	if err := c.Client.Get(ctx, req.NamespacedName, work); err != nil {
 		if apierrors.IsNotFound(err) {
-			// Cleanup derived EndpointSlices after work has been removed.
-			err = helper.DeleteEndpointSlice(c.Client, labels.Set{
```

Comment:
I think this code can be kept without deleting. 

When upgrading from a lower version to this version, this code can clean up any leftover EndpointSlices.

Reply:
Thanks, it makes sense.

Reply:
Updated, please help cc again.